stripe/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -109,62 +116,87 @@ internal class PaymentSheetViewModel internal constructor(
             runCatching {
                 stripeIntentRepository.get(args.clientSecret)
             }.fold(
-                onSuccess = ::onPaymentIntentResponse,
+                onSuccess = ::onStripeIntentResponse,
                 onFailure = {
-                    _paymentIntent.value = null
+                    _stripeIntent.value = null
                     onFatal(it)
                 }
             )
         }
     }
 
-    private fun onPaymentIntentResponse(paymentIntent: PaymentIntent) {
-        if (paymentIntent.isConfirmed) {
-            onConfirmedPaymentIntent(paymentIntent)
+    private fun onStripeIntentResponse(stripeIntent: StripeIntent) {
+        if (stripeIntent.isConfirmed) {
+            onConfirmedStripeIntent(stripeIntent)
         } else {
             runCatching {
-                paymentIntentValidator.requireValid(paymentIntent)
+                stripeIntentValidator.requireValid(stripeIntent)
             }.fold(
                 onSuccess = {
-                    _paymentIntent.value = paymentIntent
-                    resetViewState(paymentIntent)
+                    _stripeIntent.value = stripeIntent
+                    resetViewState(stripeIntent)
                 },
                 onFailure = ::onFatal
             )
         }
     }
 
     /**
-     * There's nothing left to be done in payment sheet if the PaymentIntent is confirmed.
+     * There's nothing left to be done in payment sheet if the [StripeIntent] is confirmed.
      *
      * See [How intents work](https://stripe.com/docs/payments/intents) for more details.
      */
-    private fun onConfirmedPaymentIntent(paymentIntent: PaymentIntent) {
+    private fun onConfirmedStripeIntent(stripeIntent: StripeIntent) {
         logger.info(
             """
-            PaymentIntent with id=${paymentIntent.id}" has already been confirmed.
+            ${stripeIntent.javaClass.simpleName} with id=${stripeIntent.id} has already been confirmed.
             """.trimIndent()
         )
         _viewState.value = ViewState.PaymentSheet.FinishProcessing {
             _viewState.value = ViewState.PaymentSheet.ProcessResult(
-                PaymentIntentResult(
-                    paymentIntent,
-                    StripeIntentResult.Outcome.SUCCEEDED
-                )
+                when (stripeIntent) {
+                    is PaymentIntent -> {
+                        PaymentIntentResult(
+                            stripeIntent,
+                            StripeIntentResult.Outcome.SUCCEEDED
+                        )
+                    }
+                    is SetupIntent -> {
+                        SetupIntentResult(
+                            stripeIntent,
+                            StripeIntentResult.Outcome.SUCCEEDED
+                        )
+                    }
+                    else -> throw IllegalStateException()
```

Comment:
You can also use the built-in `error(...)` and maybe add a message? `StripeIntent must be PaymentIntent or SetupIntent`?